components/RadioButtons/index.js

```diff
@@ -1,25 +1,40 @@
 import React from 'react';
 import PropTypes from 'prop-types';
-import { Text, TouchableHighlight, View, StyleSheet } from 'react-native';
+import { Text, TouchableOpacity, View, StyleSheet } from 'react-native';
 
 /**
  * Component
  */
 
-const RadioButtons = ({ buttonNames, input: { onChange, value } }) => {
+const RadioButtons = ({ buttonNames, input: { onChange, value }, mergeButtons }) => {
   const chosenIndex = buttonNames.indexOf(value);
 
   return (
-    <View style={style.buttonCollection}>
-      {buttonNames.map((button, i) => (
-        <TouchableHighlight
-          key={i}
-          style={[style.button, chosenIndex === i ? style.selectedButton : {}]}
-          onPress={() => onChange(button)}
-          underlayColor="#fff">
-          <Text style={chosenIndex === i ? style.selectedText : {}}>{button}</Text>
-        </TouchableHighlight>
-      ))}
+    <View style={mergeButtons ? style.mergedButtonCollection: style.buttonCollection}>
+      {buttonNames.map((button, i) => {
+        var styleArray = [];
+        if (mergeButtons) {
+          styleArray.push(style.mergedButton);
+          if (i == 0) {
+            styleArray.push(style.leftMergedButton);
+          } else if (i == buttonNames.length - 1) {
+            styleArray.push(style.rightMergedButton);
+          }
+        } else {
```

Comment:
nit: usually if I have >2 cases, I do a switch case instead of an if, else if, else, but not a big deal